types/ember__routing/router-service.d.ts

```diff
@@ -238,4 +239,20 @@ export default class RouterService extends Service {
         name: 'routeDidChange' | 'routeWillChange',
         callback: (transition: Transition) => void
     ): RouterService;
+
+    // https://api.emberjs.com/ember/3.12/classes/RouterService/events/routeDidChange?anchor=routeDidChange
+    // http://api.emberjs.com/ember/3.12/classes/Evented/methods/off?anchor=off
+    /**
+     * Remove a registered callback for an event.
+     *
+     * To remove the registered callback, the function you pass must be the
+     * same function as the original callback that was registered.
+     *
+     * @param name     the name of the event
+     * @param callback the same function of the original callback
+     */
+    off(
+      name: 'routeDidChange' | 'routeWillChange',
+      callback: (transition: Transition) => void
```

Comment:
Should we also support the three-argument version where the `this` context is specified as the second argument and the third parameter is the name of a method on the target? Would mean overloading with something like:
```ts
off<T extends object = object(
    name: 'routeDidChange' | 'routeWillChange',
    target: T,
    method: CallbackMethod<T>,
): RouterService;
```
Where `CalllbackMethod` is something like:
```ts
type CallbackMethod<T> = {
     [K in keyof T]: T[K] extends (transition: Transition) => void ? K : never
}[keyof T];
```
(and update `on` to support this as well.)

Reply:
Good point, thanks @jamescdavis ! I'll make that change.